src/ucp/core/ucp_ep.c

```diff
@@ -2252,21 +2252,26 @@ int ucp_ep_config_test_rndv_support(const ucp_ep_config_t *config)
            (config->key.cm_lane  != UCP_NULL_LANE);
 }
 
-unsigned ucp_ep_do_keepalive(ucp_ep_h ep)
+ucs_status_t ucp_ep_do_keepalive(ucp_ep_h ep)
 {
     ucp_lane_index_t lane;
+    ucs_status_t status;
 
     if (!ucp_ep_keepalive_is_enabled(ep)) {
-        return 0;
+        return UCS_ERR_NOT_IMPLEMENTED;
     }
 
-    /* TODO: add error handling: in future uct_ep_check may check peer
-     * immediately */
     ucs_for_each_bit(lane, ucp_ep_config(ep)->key.ep_check_map) {
         ucs_assert(lane < UCP_MAX_LANES);
         /* coverity[overrun-local] */
-        uct_ep_check(ep->uct_eps[lane], 0, NULL);
+        status = uct_ep_check(ep->uct_eps[lane], 0, NULL);
+        if (status == UCS_ERR_NO_RESOURCE) {
```

Comment:
no need to drop here, other lanes on the same ep do not depend on resources of each other

Reply:
ok, will continue for other lanes